coin/coin_test.go

```diff
@@ -439,3 +440,100 @@ func TestCoinMultiply(t *testing.T) {
 		})
 	}
 }
+
+func TestCoinDeserialization(t *testing.T) {
+	cases := map[string]struct {
+		serialized string
+		wantErr    bool
+		wantCoin   Coin
+	}{
+		"old format coin, that maps to fields directly": {
+			serialized: `{"whole": 1, "fractional": 2, "ticker": "IOV"}`,
+			wantCoin:   NewCoin(1, 2, "IOV"),
+		},
+		"old format empty coin, that maps to fields directly": {
+			serialized: `{}`,
+			wantCoin:   NewCoin(0, 0, ""),
+		},
+		"human readable format, whole without fractional": {
+			serialized: `"1IOV"`,
+			wantCoin:   NewCoin(1, 0, "IOV"),
+		},
+		"human readable format, whole without fractional, ticker space separated": {
+			serialized: `"1        IOV"`,
+			wantCoin:   NewCoin(1, 0, "IOV"),
+		},
+		"human readable format, whole and fractional": {
```

Comment:
Great tests for parsing of coin text. Works better than expected (handling spaces)